src/api/plugins/tdbp/warren_tdbp/api.py

```diff
@@ -0,0 +1,57 @@
+"""Warren API v1 tdbp router."""
+
+import logging
+from datetime import datetime
+from typing import Optional
+
+from fastapi import APIRouter, status
+from ralph.exceptions import HTTPException, LrsClientException
+
+from .indicators import ActiveActionsIndicator, ScoreIndicator
+
+router = APIRouter(
+    prefix="/tdbp",
+)
+
+logger = logging.getLogger(__name__)
+
+
+@router.get("/window")
+async def get_sliding_window(
+    until: datetime,
+    course_id: str,
+):
+    """Return course sliding window indicator."""
+    indicator = ActiveActionsIndicator(course_id=course_id, until=until)
+
+    logger.debug("Start computing 'window' indicator")
+    try:
+        results = indicator.compute()
```

Comment:
Once you've made it cache-able:

```suggestion
        results = indicator.get_or_compute()
```